### PR TITLE
Fix 1153

### PR DESCRIPTION
**Pull request by delvedor**
It might happen that `markDead` is called just after a pool update, and in such case, we will add to the dead list a node that no longer exists.
This pr fixes this issue :)
A big thanks to @penge for his initial research.

Closes: #1153 
Related: https://github.com/elastic/elasticsearch-js/issues/1122 https://github.com/elastic/elasticsearch-js/issues/1153

### CODE REVIEW

File: test/unit/connection-pool.test.js [hunk 1]
```diff
@@ -74,6 +74,14 @@ test('API', t => {
...
   })
 
+  t.test('markDead should ignore connections that no longer exists', t => {
+    const pool = new ConnectionPool({ Connection, sniffEnabled: true })
+    pool.addConnection('http://localhost:9200/')
+    pool.markDead({ id: 'foo-bar' })
+    t.deepEqual(pool.dead, [])
+    t.end()
+  })
+
```
Comment (penge) [inline test/unit/connection-pool.test.js:84]: Should be the same tested for `markAlive`?
Is it ok to `markAlive` a connection that isn't in the list of `connections`?

File: lib/pool/ConnectionPool.js [hunk 2]
```diff
@@ -60,7 +60,17 @@ class ConnectionPool extends BaseConnectionPool {
...
+      // a pool update, and in such case we will add to the dead
+      // list a node that no longer exist. The following check verify
+      // that the connection is still part of the pool before
+      // marking it as dead.
+      for (var i = 0; i < this.size; i++) {
+        if (this.connections[i].id === id) {
+          this.dead.push(id)
+          break
+        }
+      }
```
Comment (penge) [inline lib/pool/ConnectionPool.js:73]: If not added to `dead`, it still continues on to set `.status`, `.deadCount`, `.resurrectTimeout`, or call `.sort()`.
Would it be better to not do these and return `this` right after?

File: test/unit/connection-pool.test.js (hunk 1 above)
Comment (delvedor) [reply test/unit/connection-pool.test.js:84]: There's no need to add it there, as both `markAlive` and `markDead` are just modifying the given connection instance.
The only place where `markDead` is actually doing something "dangerous", is when updating the `dead` array, which is just a cache of the dead nodes.

I'm checking the existence of a connection only when modifying the `dead` array to skip useless loops :)

File: lib/pool/ConnectionPool.js (hunk 2 above)
Comment (delvedor) [reply lib/pool/ConnectionPool.js:73]: That's true, but it's not a problem as the connection will be garbage collected at the end of the function :)
There is no need to add an "exists" logic just to add an early return imho.

**Pull request merged**